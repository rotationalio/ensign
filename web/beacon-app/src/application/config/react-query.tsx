import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient();

export { QueryClientProvider, queryClient };
