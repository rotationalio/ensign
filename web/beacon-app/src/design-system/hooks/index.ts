export { default as useMenu } from './useMenu';
