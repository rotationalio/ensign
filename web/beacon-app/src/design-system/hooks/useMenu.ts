import * as React from 'react';
/**
 * React Hook to manage a menu
 *
 * It provides the logic and will be used with react context
 * to propagate its return value to all children
 *  for now we will just handle the open/close/isOpen state
 * and the anchor element
 * @param {MenuProps} props
 * @returns {MenuState}
 * @example
 * const menu = useMenu({ id: 'my-menu' })
 * return (
 * <MenuContext.Provider value={menu}>
 *  <button onClick={menu.open}>Open Menu</button>
 * <Menu id={menu.id} anchorEl={menu.anchorEl} open={menu.open}>
 * <MenuItem>Menu Item</MenuItem>
 * </Menu>
 * </MenuContext.Provider>
 * )
 */
export interface MenuProps {
  id: string;
}
export interface MenuState {
  id: string;
  anchorEl: null | HTMLElement;
  open: (event: React.MouseEvent<HTMLElement>) => void;
  close: () => void;
  isOpen: boolean;
}
const useMenu = (props: MenuProps): MenuState => {
  const { id } = props;
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };
  const close = () => {
    setAnchorEl(null);
  };

  const isOpen = Boolean(anchorEl);

  return { id, anchorEl, open, close, isOpen };
};

export default useMenu;
