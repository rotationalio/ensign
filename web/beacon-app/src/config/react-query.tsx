import { MutationCache, QueryCache, QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient({
  // define default options for all queries
  defaultOptions: {
    queries: {
      staleTime: 0,
      cacheTime: 5 * 60 * 1000,
      retry: 3,
      retryDelay: (attemptIndex) => Math.min(1000 * 2 ** attemptIndex, 30000),
      refetchInterval: false,
      refetchIntervalInBackground: false,
      refetchOnMount: true,
      refetchOnReconnect: true,
      refetchOnWindowFocus: true,
    },
  },
});

const queryCache = new QueryCache({
  onError: (error) => {
    console.log(error);
  },
  onSuccess: (data) => {
    console.log(data);
  },
});
const mutationCache = new MutationCache({
  onError: (error) => {
    console.log(error);
  },
  onSuccess: (data) => {
    console.log(data);
  },
});
export { mutationCache, queryCache, queryClient, QueryClientProvider };
