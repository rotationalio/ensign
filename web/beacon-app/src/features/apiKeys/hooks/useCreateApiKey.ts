import { useMutation } from '@tanstack/react-query';

import axiosInstance from '@/application/api/ApiService';
import { queryClient } from '@/application/config/react-query';
import { RQK } from '@/constants';

import { createProjectAPIKey } from '../api/createApiKey';
import { APIKeyMutation } from '../types/createApiKeyService';
export function useCreateProjectAPIKey(): APIKeyMutation {
  const mutation = useMutation(createProjectAPIKey(axiosInstance), {
    retry: 0,
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: [RQK.PROJECT_API_KEYS] });
      queryClient.invalidateQueries({ queryKey: [RQK.QUICK_VIEW] });
      queryClient.invalidateQueries({ queryKey: [RQK.API_KEYS] });
      queryClient.invalidateQueries({ queryKey: [RQK.PROJECT_QUICK_VIEW] });
      queryClient.invalidateQueries({ queryKey: [RQK.PROJECTS] });
      queryClient.invalidateQueries({ queryKey: [RQK.PROJECT] });
    },
  });
  return {
    createProjectNewKey: mutation.mutate,
    reset: mutation.reset,
    key: mutation.data as APIKeyMutation['key'],
    hasKeyFailed: mutation.isError,
    wasKeyCreated: mutation.isSuccess,
    isCreatingKey: mutation.isLoading,
    error: mutation.error,
  };
}
