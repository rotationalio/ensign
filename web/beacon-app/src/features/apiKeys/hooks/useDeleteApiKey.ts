import { useMutation } from '@tanstack/react-query';

import axiosInstance from '@/application/api/ApiService';
import { queryClient } from '@/application/config/react-query';
import { RQK } from '@/constants';

import { deleteAPIKeyRequest } from '../api/deleteApiKeyApi';
import { DeleteAPIKeyMutation } from '../types/deleteApiKeyService';
export function useDeleteAPIKey(apiKey: string): DeleteAPIKeyMutation {
  const mutation = useMutation(() => deleteAPIKeyRequest(axiosInstance)(apiKey), {
    retry: 0,
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: [RQK.PROJECT_API_KEYS] });
      queryClient.invalidateQueries({ queryKey: [RQK.QUICK_VIEW] });
      queryClient.invalidateQueries({ queryKey: [RQK.API_KEYS] });
      queryClient.invalidateQueries({ queryKey: [RQK.PROJECT_QUICK_VIEW] });
      queryClient.invalidateQueries({ queryKey: [RQK.PROJECTS] });
      queryClient.invalidateQueries({ queryKey: [RQK.PROJECT] });
    },
  });
  return {
    deleteApiKey: mutation.mutate,
    reset: mutation.reset,
    key: mutation.data as DeleteAPIKeyMutation['key'],
    hasKeyDeletedFailed: mutation.isError,
    wasKeyDeleted: mutation.isSuccess,
    isDeletingKey: mutation.isLoading,
    error: mutation.error,
  };
}
