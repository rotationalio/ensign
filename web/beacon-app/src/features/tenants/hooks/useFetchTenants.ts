import { useQuery } from '@tanstack/react-query';

import axiosInstance from '@/application/api/ApiService';
import { RQK } from '@/constants/queryKeys';

import { tenantsRequest } from '../api/tenantListAPI';
import { TenantsQuery } from '../types/tenantServices';

export function useFetchTenants(): TenantsQuery {
  const query = useQuery([RQK.TENANTS], tenantsRequest(axiosInstance), {
    refetchOnWindowFocus: false,
    refetchOnMount: true,
    // set stale time to 15 minutes
    // TODO: Change stale time
    staleTime: 1000 * 60 * 15,
  });

  return {
    getTenants: query.refetch,
    hasTenantsFailed: query.isError,
    isFetchingTenants: query.isLoading,
    tenants: query.data,
    wasTenantsFetched: query.isSuccess,
    error: query.error,
  };
}
