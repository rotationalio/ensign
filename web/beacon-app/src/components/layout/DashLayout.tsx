import { t } from '@lingui/macro';
import invariant from 'invariant';
import { useEffect } from 'react';

import { useFetchTenants } from '@/features/tenants/hooks/useFetchTenants';

// import { useOrgStore } from '@/store';
import { MainStyle } from './DashLayout.styles';
import MobileFooter from './MobileFooter';
import { Sidebar } from './Sidebar';

type DashLayoutProps = {
  children?: React.ReactNode;
};

const DashLayout: React.FC<DashLayoutProps> = ({ children }) => {
  // const Store = useOrgStore.getState() as any;
  const { tenants, wasTenantsFetched } = useFetchTenants();
  const hasTenants =
    tenants?.tenants && Array.isArray(tenants?.tenants) && tenants?.tenants?.length > 0;
  // const tenantID = tenants?.tenants[0]?.id;

  // ensure the tenant is loaded correctly otherwise we need to throw an error
  useEffect(() => {
    if (wasTenantsFetched) {
      invariant(
        hasTenants,
        t`Something went wrong. Please contact us at support@rotational.io for assistance.`
      );
    }
  }, [hasTenants, wasTenantsFetched]);

  // useEffect(() => {
  //   if (wasTenantsFetched && tenantID) {
  //     Store.setTenantID(tenantID);
  //   }
  // }, [tenantID, wasTenantsFetched, Store]);

  return (
    <div className="flex flex-col md:pl-[250px]">
      <Sidebar className="hidden md:block" />
      <MainStyle>{children}</MainStyle>
      <MobileFooter />
    </div>
  );
};

export default DashLayout;
