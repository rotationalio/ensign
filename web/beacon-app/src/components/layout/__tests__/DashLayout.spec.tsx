/* eslint-disable testing-library/await-async-utils */
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render, screen, waitFor } from '@testing-library/react';
import { vi } from 'vitest';

import * as useFetchTenants from '@/features/tenants/hooks/useFetchTenants';

import DashLayout from '../DashLayout';

const queryClient = new QueryClient();

const renderDashLayout = () => {
  return render(
    <QueryClientProvider client={queryClient}>
      <DashLayout />
    </QueryClientProvider>
  );
};
vi.mock('invariant');
// mock t tag from lingui
vi.mock('@lingui/macro', () => ({
  t: (str) => str,
  Trans: ({ children }) => children,
}));

// mock useFetchTenants hook
vi.mock('@/features/tenants/hooks/useFetchTenants', () => ({
  __esModule: true,
  useFetchTenants: () => ({
    tenants: {
      tenants: [
        {
          id: 1,
          name: 'test',
        },
      ],
    },
    wasTenantsFetched: true,
  }),
}));

// mock useDropdownMenu
vi.mock('@/components/MenuDropdown/useDropdownMenu', () => ({
  __esModule: true,
  useDropdownMenu: () => ({
    menuItems: [
      {
        label: 'test',
        value: 'test',
      },
    ],
  }),
}));

// mock router hooks
vi.mock('react-router-dom', () => ({
  useNavigate: () => vi.fn(),
  Link: ({ children }) => children,
  useLocation: () => ({
    pathname: '/test',
  }),
  NavLink: ({ children }) => children,
}));

// mock useOrgStore
vi.mock('@/store', () => ({
  __esModule: true,
  useOrgStore: {
    getState: () => ({
      org: {
        name: 'test',
      },
      setTenantID: vi.fn(),
      setOrgName: vi.fn(),
    }),
  },
}));

describe('DashLayout', () => {
  beforeEach(() => {
    vi.resetAllMocks();
  });

  it('should render the component', () => {
    const { container } = renderDashLayout();
    expect(container).toMatchSnapshot();
  });

  it('should render the error component when the tenant is not loaded correctly', () => {
    vi.spyOn(useFetchTenants, 'useFetchTenants')
      .mockImplementation()
      .mockReturnValue({
        tenants: {
          tenants: null,
        },
        wasTenantsFetched: true,
      });

    renderDashLayout();

    waitFor(() => {
      // check if testid is rendered
      expect(
        screen.getByText(
          'Something went wrong. Please contact us at support@rotational.io for assistance.'
        )
      );
    });
  });
});
