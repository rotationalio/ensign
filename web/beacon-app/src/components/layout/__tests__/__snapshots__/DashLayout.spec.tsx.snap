// Vitest Snapshot v1

exports[`DashLayout > should render the component 1`] = `
<div>
  <div
    class="flex flex-col md:pl-[250px]"
  >
    <aside
      class="fixed top-0 left-0 flex h-screen flex-col bg-[#1D65A6] pt-5 pb-10 text-white md:w-[250px] hidden md:block"
    >
      <div
        class="flex h-full flex-col"
      >
        <div
          class="grow"
        >
          <div
            aria-hidden="true"
            class="flex w-full flex-row items-center justify-between py-2 pr-5 pl-8 text-sm outline-none"
            data-testid="menu"
            role="button"
            tabindex="0"
          >
            <div
              class="flex items-center gap-3 "
            >
              <span
                class="sc-hHTYSt cKmJRb"
              />
              <h1
                class="flex"
                data-testid="orgName"
              >
                <svg
                  class="sc-dkrFOg fRpLxR flex"
                  viewBox="0 0 50 50"
                >
                  <circle
                    class="path"
                    cx="25"
                    cy="25"
                    fill="none"
                    r="20"
                    stroke-width="2"
                  />
                </svg>
                <br />
              </h1>
            </div>
            <div
              class="flex-end"
            >
              <div
                class="relative"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="menu"
                  data-state="closed"
                  id="radix-:r0:"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="6"
                    viewBox="0 0 11 6"
                    width="11"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.6635 0.336517C10.9719 0.644826 10.9719 1.14469 10.6635 1.453L6.45302 5.66353C6.14471 5.97184 5.64484 5.97184 5.33653 5.66353L1.12601 1.453C0.817699 1.14469 0.817699 0.644826 1.12601 0.336517C1.43432 0.0282085 1.93418 0.0282085 2.24249 0.336517L5.89478 3.9888L9.54706 0.336517C9.85537 0.0282085 10.3552 0.0282085 10.6635 0.336517Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="pt-8"
          >
            <div>
              <div
                class="mb-2 min-h-[8px] list-none text-white last:mb-0"
              >
                <span
                  class="relative z-[1] mr-3 w-[24px] text-white"
                >
                  <svg
                    fill="currentColor"
                    font-size="24"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 0 0-44.4 0L77.5 505a63.9 63.9 0 0 0-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0 0 18.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                    />
                  </svg>
                </span>
                <span
                  class="relative z-[1] flex font-normal"
                >
                  Home
                   
                </span>
              </div>
              <div
                class="mb-2 min-h-[8px] list-none text-white last:mb-0"
              >
                <span
                  class="relative z-[1] mr-3 w-[24px] text-white"
                >
                  <svg
                    fill="currentColor"
                    font-size="24"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M280 752h80c4.4 0 8-3.6 8-8V280c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8v464c0 4.4 3.6 8 8 8zm192-280h80c4.4 0 8-3.6 8-8V280c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8v184c0 4.4 3.6 8 8 8zm192 72h80c4.4 0 8-3.6 8-8V280c0-4.4-3.6-8-8-8h-80c-4.4 0-8 3.6-8 8v256c0 4.4 3.6 8 8 8zm216-432H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
                    />
                  </svg>
                </span>
                <span
                  class="relative z-[1] flex font-normal"
                >
                  Projects
                   
                </span>
              </div>
              <div
                class="mb-2 min-h-[8px] list-none text-white last:mb-0"
              >
                <span
                  class="relative z-[1] mr-3 w-[24px] text-white"
                >
                  <svg
                    fill="currentColor"
                    font-size="24"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M824.2 699.9a301.55 301.55 0 0 0-86.4-60.4C783.1 602.8 812 546.8 812 484c0-110.8-92.4-201.7-203.2-200-109.1 1.7-197 90.6-197 200 0 62.8 29 118.8 74.2 155.5a300.95 300.95 0 0 0-86.4 60.4C345 754.6 314 826.8 312 903.8a8 8 0 0 0 8 8.2h56c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5A226.62 226.62 0 0 1 612 684c60.9 0 118.2 23.7 161.3 66.8C814.5 792 838 846.3 840 904.3c.1 4.3 3.7 7.7 8 7.7h56a8 8 0 0 0 8-8.2c-2-77-33-149.2-87.8-203.9zM612 612c-34.2 0-66.4-13.3-90.5-37.5a126.86 126.86 0 0 1-37.5-91.8c.3-32.8 13.4-64.5 36.3-88 24-24.6 56.1-38.3 90.4-38.7 33.9-.3 66.8 12.9 91 36.6 24.8 24.3 38.4 56.8 38.4 91.4 0 34.2-13.3 66.3-37.5 90.5A127.3 127.3 0 0 1 612 612zM361.5 510.4c-.9-8.7-1.4-17.5-1.4-26.4 0-15.9 1.5-31.4 4.3-46.5.7-3.6-1.2-7.3-4.5-8.8-13.6-6.1-26.1-14.5-36.9-25.1a127.54 127.54 0 0 1-38.7-95.4c.9-32.1 13.8-62.6 36.3-85.6 24.7-25.3 57.9-39.1 93.2-38.7 31.9.3 62.7 12.6 86 34.4 7.9 7.4 14.7 15.6 20.4 24.4 2 3.1 5.9 4.4 9.3 3.2 17.6-6.1 36.2-10.4 55.3-12.4 5.6-.6 8.8-6.6 6.3-11.6-32.5-64.3-98.9-108.7-175.7-109.9-110.9-1.7-203.3 89.2-203.3 199.9 0 62.8 28.9 118.8 74.2 155.5-31.8 14.7-61.1 35-86.5 60.4-54.8 54.7-85.8 126.9-87.8 204a8 8 0 0 0 8 8.2h56.1c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5 29.4-29.4 65.4-49.8 104.7-59.7 3.9-1 6.5-4.7 6-8.7z"
                    />
                  </svg>
                </span>
                <span
                  class="relative z-[1] flex font-normal"
                >
                  Team
                   
                </span>
              </div>
            </div>
            <hr
              class="my-5 mx-8"
            />
            <div>
              <div
                class="mb-2 min-h-[8px] list-none text-white last:mb-0"
              >
                <span
                  class="relative z-[1] mr-3 w-[24px] text-white"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    font-size="24"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 21h10a2 2 0 002-2V9.414a1 1 0 00-.293-.707l-5.414-5.414A1 1 0 0012.586 3H7a2 2 0 00-2 2v14a2 2 0 002 2z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </span>
                <span
                  class="relative z-[1] flex font-normal"
                >
                  Docs
                   
                  <svg
                    class="ml-1 h-3 w-3 text-white"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.5 6H5.25A2.25 2.25 0 003 8.25v10.5A2.25 2.25 0 005.25 21h10.5A2.25 2.25 0 0018 18.75V10.5m-10.5 6L21 3m0 0h-5.25M21 3v5.25"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="mb-2 min-h-[8px] list-none text-white last:mb-0"
              >
                <a
                  class="flex h-12 items-center whitespace-nowrap pl-8 text-sm"
                  href="mailto:support@rotational.io"
                >
                  <span
                    class="relative z-[1] mr-3 w-[24px] text-white"
                  >
                    <svg
                      fill="currentColor"
                      font-size="24"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0V0z"
                        fill="none"
                      />
                      <path
                        d="M11 23.59v-3.6c-5.01-.26-9-4.42-9-9.49C2 5.26 6.26 1 11.5 1S21 5.26 21 10.5c0 4.95-3.44 9.93-8.57 12.4l-1.43.69zM11.5 3C7.36 3 4 6.36 4 10.5S7.36 18 11.5 18H13v2.3c3.64-2.3 6-6.08 6-9.8C19 6.36 15.64 3 11.5 3zm-1 11.5h2v2h-2zm2-1.5h-2c0-3.25 3-3 3-5 0-1.1-.9-2-2-2s-2 .9-2 2h-2c0-2.21 1.79-4 4-4s4 1.79 4 4c0 2.5-3 2.75-3 5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="relative z-[1] flex"
                  >
                    Support
                    <svg
                      class="ml-1 h-3 w-3 text-white"
                      fill="none"
                      stroke="currentColor"
                      stroke-width="1.5"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13.5 6H5.25A2.25 2.25 0 003 8.25v10.5A2.25 2.25 0 005.25 21h10.5A2.25 2.25 0 0018 18.75V10.5m-10.5 6L21 3m0 0h-5.25M21 3v5.25"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                  </span>
                </a>
              </div>
              <div
                class="mb-2 min-h-[8px] list-none text-white last:mb-0"
              >
                <span
                  class="relative z-[1] mr-3 w-[24px] text-white"
                >
                  <svg
                    fill="none"
                    font-size="24"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M16 9C16 11.2091 14.2091 13 12 13C9.79086 13 8 11.2091 8 9C8 6.79086 9.79086 5 12 5C14.2091 5 16 6.79086 16 9ZM14 9C14 10.1046 13.1046 11 12 11C10.8954 11 10 10.1046 10 9C10 7.89543 10.8954 7 12 7C13.1046 7 14 7.89543 14 9Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM3 12C3 14.0902 3.71255 16.014 4.90798 17.5417C6.55245 15.3889 9.14627 14 12.0645 14C14.9448 14 17.5092 15.3531 19.1565 17.4583C20.313 15.9443 21 14.0524 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12ZM12 21C9.84977 21 7.87565 20.2459 6.32767 18.9878C7.59352 17.1812 9.69106 16 12.0645 16C14.4084 16 16.4833 17.1521 17.7538 18.9209C16.1939 20.2191 14.1881 21 12 21Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="relative z-[1] flex font-normal"
                >
                  Profile
                   
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ml-8 space-y-3"
        >
          <ul
            class="space-y-1 text-xs text-white"
          >
            <li>
              About
               
              <svg
                class="ml-1 h-3 w-3 text-white"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5 6H5.25A2.25 2.25 0 003 8.25v10.5A2.25 2.25 0 005.25 21h10.5A2.25 2.25 0 0018 18.75V10.5m-10.5 6L21 3m0 0h-5.25M21 3v5.25"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </li>
            <li>
              Contact Us
               
              <svg
                class="ml-1 h-3 w-3 text-white"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5 6H5.25A2.25 2.25 0 003 8.25v10.5A2.25 2.25 0 005.25 21h10.5A2.25 2.25 0 0018 18.75V10.5m-10.5 6L21 3m0 0h-5.25M21 3v5.25"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </li>
            <li>
              Server Status
               
              <svg
                class="ml-1 h-3 w-3 text-white"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5 6H5.25A2.25 2.25 0 003 8.25v10.5A2.25 2.25 0 005.25 21h10.5A2.25 2.25 0 0018 18.75V10.5m-10.5 6L21 3m0 0h-5.25M21 3v5.25"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </li>
            <li>
              © Rotational Labs, Inc
               
              <svg
                class="ml-1 h-3 w-3 text-white"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5 6H5.25A2.25 2.25 0 003 8.25v10.5A2.25 2.25 0 005.25 21h10.5A2.25 2.25 0 0018 18.75V10.5m-10.5 6L21 3m0 0h-5.25M21 3v5.25"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </li>
          </ul>
          <p
            class="w-full text-xs text-white"
          >
            <span>
              App Version 
              0.0.0
               
            </span>
            <span>
              & Git Revision 
              0.0.0
               
            </span>
          </p>
        </div>
      </div>
    </aside>
    <main
      class="sc-bcXHqe a-dwtT"
    />
    <footer
      class="fixed bottom-0 left-0 w-screen bg-[#1D65A6] py-1 md:hidden md:pl-[250px]"
    >
      <ul
        class="flex items-center justify-around space-y-1 text-xs text-white"
      >
        <li>
          About
        </li>
        <li>
          Contact Us
        </li>
        <li>
          Server Status
        </li>
        <li>
          © Rotational Labs, Inc
        </li>
      </ul>
    </footer>
  </div>
</div>
`;
